VideoLocker/src/androidTest/java/org/edx/mobile/test/feature/data/Credentials.java

```diff
@@ -1,11 +1,30 @@
 package org.edx.mobile.test.feature.data;
 
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.edx.mobile.util.Config;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.util.UUID;
+
 public class Credentials {
     public final String email;
     public final String password;
+    @Nullable public String username;
 
     public Credentials(String email, String password) {
         this.email = email;
         this.password = password;
     }
+
+    public static Credentials freshCredentials(@NonNull Config config) {
+        String randomUdid = UUID.randomUUID().toString();
+        String testEmail = ResourceUtil.getFormattedString(
+                config.getEndToEndConfig().getEmailTemplate(),
+                "unique_id", randomUdid).toString();
+        Credentials creds = new Credentials(testEmail, randomUdid);
+        creds.username = testEmail.split("@")[0];
```

Comment:
This assignment should be done in the constructor, in which case the `username` field could also be defined as `final` and `@NonNull` (the other fields and parameters can also have the `@NonNull` annotation if you like).